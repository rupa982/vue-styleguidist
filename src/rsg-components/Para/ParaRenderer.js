import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

export const styles = ({ space, fontFamily, color }) => ({
	para: {
		marginTop: 0,
		marginBottom: space[2],
		color: color.base,
		fontFamily: fontFamily.base,
		fontSize: 'inherit',
		lineHeight: 1.5,
	},
});

export function ParaRenderer({ classes, children }) {
	return <div className={classes.para}>{children}</div>;
}

ParaRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node.isRequired,
};

export default Styled(styles)(ParaRenderer);
