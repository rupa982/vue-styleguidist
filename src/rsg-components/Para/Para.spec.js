import React from 'react';
import { ParaRenderer } from './ParaRenderer';

it('should render paragraph', () => {
	const actual = shallow(<ParaRenderer classes={{}}>Pizza</ParaRenderer>);

	expect(actual).toMatchSnapshot();
});
