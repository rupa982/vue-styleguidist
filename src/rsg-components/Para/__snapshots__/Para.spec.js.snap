// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render paragraph 1`] = `
<div>
  Pizza
</div>
`;
